%YAML:1.2

#--------------------------------------------------------------------------------------------
# Camera params. Pinhole
#--------------------------------------------------------------------------------------------
cam_rgb_topic_: /kinect2/qhd/image_color
cam_depth_topic_: /kinect2/qhd/image_depth_rect

cam_fx_: 5.3701347143784051e+02
cam_fy_: 5.4031150386085142e+02
cam_cx_: 4.8090420414824069e+02
cam_cy_: 2.7061959759552622e+02
cam_k1_: 4.3949139689180149e-02
cam_k2_: -3.8480936124160148e-02
cam_p1_: -1.1869517130142055e-03
cam_p2_: 1.4894437810783877e-03
#cam_k3_: 0
cam_k3_: 0.01110263483766991

cam_height_: 540
cam_width_: 960

cam_depth_factor_: 0.001   # Depth scale factor.

cam_dmax_: 8.0   # Max depth value to be used. (m)
cam_dmin_: 0.1   # Min depth value to be used. (m) 
 
cam_fps_: 20     # Camera FPs.

#--------------------------------------------------------------------------------------------
# Robot intrinsic and extrinsic
#--------------------------------------------------------------------------------------------
#### Intrinsic
encoder_topic_: /sensor_state

#odom_kl_: 4.0652e-5   # left wheel factor
#odom_kr_: 4.0668e-5   # right wheel factor
odom_kl_: 5.0684e-5   # left wheel factor
odom_kr_: 5.0684e-5   # right wheel factor
odom_b_: 0.287       # wheel space
odom_K_: 0.008        # Noise factor.

#### Extrinsic Trc
Trc_: !!opencv-matrix
   rows: 3
   cols: 4
   dt: d
   data: [   0.0, 0.104528463267653, 0.994521895368273, 0.0, -1.0, 0.0, 0.0, 0.0, 0.0, -0.994521895368273, 0.104528463267653, 0.05 ]
#--------------------------------------------------------------------------------------------
# RGB-D Encoder Tracking
#--------------------------------------------------------------------------------------------
#### ORB feature 
ret_ft_n_features_: 2000   # Number of ORB features per frame.

#### Tracking
ret_tk_dist_th_: 4.0    # Local map search radius. (m)
ret_tk_angle_th_: 1.0   # Local map search angle. (rad)
ret_tk_db_: 0.20         # Base threshold for erroneous match discard. (m)
ret_tk_kd_: 0.025        # Scale factor of the threshold for erroneous match discard.

#### Keyframe decision
# Condation 1.
ret_kd_fps_factor_: 0.9
# Condation 2: 
ret_kd_dist_th_: 0.2    # Max distance (m)
ret_kd_angle_th_: 0.4   # Max angle (rad)

#--------------------------------------------------------------------------------------------
# Dynamic Pixels Culling
#--------------------------------------------------------------------------------------------
dpc_n_near_kfs_: 5                # Number of near keyframes.
dpc_npts_per_cluster_: 6000       # Number of points per cluster.
dpc_n_sel_pts_per_cluster_: 100   # Number of points per cluster to be selected for dynamic cluster decision.
dpc_search_square_size_: 9        # 9 pixels

#--------------------------------------------------------------------------------------------
# Extend by depth parameters
#--------------------------------------------------------------------------------------------
#depth_thread_: 0.008
depth_thread_: 0.1
depth_thread_second_: 0.0001
aera_thread_: 100
blur_kernal_size_: 5
hight_limit_: 30

#--------------------------------------------------------------------------------------------
# Sparse Mapping 
#--------------------------------------------------------------------------------------------
#### Local Mapping
sm_lm_window_size_: 8  # local BA window size sp_lm_window_size_ KFs.


#--------------------------------------------------------------------------------------------
# OctoMap Construction
#--------------------------------------------------------------------------------------------
oc_voxel_size_: 0.05   # Voxel size of the OctoMap (m).
oc_submap_size_: 5    # Sub-OctoMap size (KFs)

#OctoMap parameters 
#oc_occ_th_ : 0.61
#oc_prob_hit_ : 0.6
#oc_prob_miss_ : 0.45
oc_occ_th_ : 0.61
oc_prob_hit_ : 0.53
oc_prob_miss_ : 0.45

